ait/dsn/proc/deframe_packet_processor.py

```diff
@@ -0,0 +1,1167 @@
+  	output port of the upstream service (AIT DSN RAF/RCF server?)
+   	"""
+    def __init__(self, framequeue, *args, **kwargs):
+        """
+        Constructor
+        :param framequeue: The frame queue to which frames from ports will be added
+        :param args: Arguments
+        :param kwargs: Keyword arguments
+        """
+
+        self._frame_queue = framequeue
+
+        # Incoming port for the frames from DSN services
+        self._listening_host = '127.0.0.1'
+        self._listening_port = int(ait.config.get('dsn.sle.frame_output_port',
+                                                    kwargs.get('frame_output_port',
+                                                               ait.DEFAULT_FRAME_PORT)))
+
+        # Frame type expected from the DSN services
+        self._downlink_frame_type = ait.config.get('dsn.sle.downlink_frame_type',
+                                                   kwargs.get('downlink_frame_type',
+                                                              ait.DEFAULT_FRAME_TYPE))
+
+        # Grab the class from frames only once
+        self._tm_frame_class = getattr(frames, self._downlink_frame_type)
+
+        # Inform server of the address to listen
+        listener = (self._listening_host, self._listening_port)
+        super(Frame_Service, self).__init__(listener)
+
+    def handle(self, data, address):
+        """
+        This handler is called whenever a message is received by the DatagramServer;
+        capturing that message data and putting it into the field queue for downstream
+        processing
+        :param data: The data of the message received
+        :param address: The address that sent the message
+        """
+
+        in_frame = self._tm_frame_class(data)
+
+        # Add any frame-based logic/decisions here
+        if in_frame.is_idle_frame:
+            log.debug('Dropping {} marked as an idle frame'.format(self._tm_frame_class))
+            return
+
+        self._frame_queue.append( ( in_frame, ( int(time.time())) ) )
+
+    def start(self):
+        """Starts this Frame_Service."""
+        values = self._downlink_frame_type, self._listening_host, self._listening_port
+        log.info('Listening for %s frames on %s:%d (UDP)' % values)
+        super(Frame_Service, self).start()
+
+    def stop(self):
+        """Stop this Frame_Service."""
+        values = self._downlink_frame_type, self._listening_host, self._listening_port
+        log.info('Stop listener for %s frames on %s:%d (UDP)' % values)
+        super(Frame_Service, self).stop(timeout=5)
+
+if __name__ == '__main__':
+
+    processor = Processor()
+    try:
+        processor.run()
+        while True:
+            ait.core.log.info('Sleeping...')
+            gevent.sleep(1)
+    except KeyboardInterrupt:
+        print('Disconnecting...')
+    except Exception as e:
+        print(traceback.print_exc())
+    finally:
+        processor.stop()
+        time.sleep(2)
```

Comment:
Can we move this to a script in `ait/dsn/bin` so it's consistent with the other bin scripts in DSN and the other repos?